scripts/install-tools.sh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# renovate: datasource=github-releases depName=kubernetes-sigs/controller-tools
+CONTROLLER_TOOLS_GENERATOR_VERSION=v0.15.0
+# renovate: datasource=github-releases depName=golangci/golangci-lint
+GOLANG_CI_LINTER_VERSION=v2.0.2
+
+# Execute
+# scripts/install-tools.sh
+# scripts/install-tools.sh -h
+# scripts/install-tools.sh --generator
+# scripts/install-tools.sh --golangci
+
+show_help() {
+cat << EOF
+'external-dns' helm linter helper commands
+
+Usage: $(basename "$0") <options>
+    -h, --help          Display help
+    --generator         Install generator
+    --golangci         Install golangci linter
+EOF
+}
+
+install_generator() {
+  # https://github.com/kubernetes-sigs/controller-tools/blob/main/cmd/controller-gen/main.go
+  local install=false
+  if [[ -x $(which controller-gen) ]]; then
+      local version=$(controller-gen --version | sed 's/Version: //')
+      if [[ "${version}" == "${CONTROLLER_TOOLS_GENERATOR_VERSION}" ]]; then
+          install=false
+        else
+          install=true
+      fi
+    else
+      install=true
+  fi
+  if [[ "$install" == true ]]; then
+      set -ex ;\
+	    go install sigs.k8s.io/controller-tools/cmd/controller-gen@${CONTROLLER_TOOLS_GENERATOR_VERSION} ;
+  fi
+}
+
+install_golangci() {
+  local install=false
+  if [[ -x $(which golangci-lint) ]]; then
+      local version=$(golangci-lint version --short)
+      if [[ "${version}" == "${GOLANG_CI_LINTER_VERSION#v}" ]]; then
+          install=false
+        else
+          install=true
+      fi
+    else
+      install=true
+  fi
+  if [[ "$install" == true ]]; then
+      curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/HEAD/install.sh \
```

Comment:
do we want to pin version here?

Reply:
It's pinned on line 20 https://github.com/kubernetes-sigs/external-dns/blob/aa7f5ca0d7ce91f67d18b0bdca875e66b092e451/scripts/install-tools.sh#L20C1-L20C32

this script is not versioned to my understanding, but version is passed to the script.

Docs how to install https://golangci-lint.run/welcome/install/

Reply:
Yeah I would still replace `HEAD` with a commit so that we are not subject to weird changes to the script that either break things or worse are a security risk.

Reply:
added a tag, as commit sha is trictky to manage. will this work?

Reply:
Why is it tricky? It would just be `https://raw.githubusercontent.com/golangci/golangci-lint/cc3567e3127d8530afb69be1b7bd20ba9ebcc7c1/install.sh`.

Reply:
Automated updates tricky

Reply:
Changed to commit sha